packages/babel-parser/src/parser/expression.js

```diff
@@ -1854,37 +1855,47 @@ export default class ExpressionParser extends LValParser {
   // Parse template expression.
 
   parseTemplateElement(isTagged: boolean): N.TemplateElement {
-    const elem = this.startNode();
-    if (this.state.value === null) {
+    const { start, end, value } = this.state;
+    const elemStart = start + 1;
+    const elem = this.startNodeAt(
+      elemStart,
+      createPositionFromPosition(this.state.startLoc, 1),
```

Comment:
Nit: it would be more clear if this function was called `createPositionAfterOffset` or `getIncrementedPosition`

Reply:
Maybe `createPositionWithColumnOffset`? I should add a note that this function does not consider workspace, so it does not work for any offsets for sure, and thus it should be only used when we create AST node inside the token boundaries, which should be rare scenario.

For this reason, I am also tempted to expand `TemplateElement` to match the actual token range in Babel 8. It can possibly further get rid of https://github.com/babel/babel/blob/531db5dc4e3a0baa2c731a520c48bb490f1845a9/eslint/babel-eslint-parser/src/convert/convertAST.cjs#L79-L106